pkg/dataobj/internal/dataset/column_builder.go

```diff
@@ -73,6 +91,10 @@ type ColumnBuilder struct {
 // provided options. NewColumnBuilder returns an error if the options are
 // invalid.
 func NewColumnBuilder(tag string, opts BuilderOptions) (*ColumnBuilder, error) {
+	if opts.Compression == datasetmd.COMPRESSION_TYPE_ZSTD && opts.CompressionOptions.ZstdWriter == nil {
```

Comment:
OK, I added this to make it easier for the end user to init a zstd compressed column, but this will actually create a single zstd encoder per column because this late init happens after the options are passed into various column builders. Instead I will return an error here and force the user to define the right options.